src/main/java/org/scoverage/plugin/SCoverageCheckMojo.java

```diff
@@ -168,31 +223,27 @@ public void execute() throws MojoFailureException
         int invokedBranchesCount = coverage.invokedBranchesCount();
         int invokedStatementCount = coverage.invokedStatementCount();
 
-        getLog().info( String.format( "Statement coverage.: %s%%", coverage.statementCoverageFormatted() ) );
-        getLog().info( String.format( "Branch coverage....: %s%%", coverage.branchCoverageFormatted() ) );
         getLog().debug( String.format( "invokedBranchesCount:%d / branchCount:%d, invokedStatementCount:%d / statementCount:%d",
                                       invokedBranchesCount, branchCount, invokedStatementCount, statementCount ) );
-        if ( minimumCoverage > 0.0 )
+
+        boolean ok = checkCoverage( getLog(), "Total", coverage,
+                                    minimumCoverage, minimumCoverageBranchTotal, true );
+        for ( MeasuredPackage pkgCoverage : CollectionConverters.asJava( coverage.packages() ) )
```

Comment:
this could be a large list, right? for files it will be even larger
 how about wrapping this into `if(minimumCoverageStmtPerFile > 0.0 && minimumCoverageBranchPerPackage > 0.0)` to avoid checking every package/file with default/0 minimum?

Reply:
will do